src/com/esotericsoftware/kryo/util/Util.java

```diff
@@ -26,6 +26,9 @@
 /** A few utility methods, mostly for private use.
  * @author Nathan Sweet <misc@n4te.com> */
 public class Util {
+	//See: https://stackoverflow.com/questions/3038392/do-java-arrays-have-a-maximum-size
+	static public final int maxSafeArraySize = Integer.MAX_VALUE - 8;

```

Comment:
As a constant it should probably follow naming conventions for constants?
Alternatively, this could be defined somewhere in the io package and be package private, with the advantage that the public api is not extended (but keeping it here is fine for me as well).